casadm/cas_main.c

```diff
@@ -154,6 +154,84 @@ int command_handle_option(char *opt, const char **arg)
 	return SUCCESS;
 }
 
+/* Filler to print sub-commands */
+int cmd_subcmd_print_subcmd(cli_option* option, int flag)
+{
+	return !!(option->flags & CLI_OPTION_SUBCMD);
+}
+
+/* Filler to print parameters of given sub-command */
+int cmd_subcmd_print_param(cli_option* option, int flag)
+{
+	return (flag == (option->priv & flag));
+}
+
+static inline void cmd_subcmd_print_invalid_subcmd(cli_option* options)
+{
+	cas_printf(LOG_ERR, "Invalid or missing first sub-command parameter. ");
+	cas_printf(LOG_ERR, "Expected one of the: {");
+	print_options_usage(options, "|", cmd_subcmd_print_subcmd, 0);
+	cas_printf(LOG_ERR, "}\n");
+}
+
+/* Print help for command with subcommands */
+void cmd_subcmd_help(app *app_values, cli_command *cmd, int flag_required)
```

Comment:
would it be much work to extend submcd help logic to only print help for one specific subcommand if user provided it via commandline? 

For example, to print a shorter help when user types
casadm --standby --init --help

Reply:
Yeah, the `-H` or `--help` flag is detected at the very early stage of command parsing and it effectively interrupts the parsing to print per-command help. Increasing granularity would require changes in the parser (and probably including idea of the subcmd in the parser itself, because right now subcmds are implemented with kind of workaround).